### ISSUE TITLE
Can't connect to influxdb - ui on localhost:8083 not available

### ISSUE THREAD

**Issue by ezcocos** [influxdb/influxdb#1164]
Dear All,
I have already raised this issue in another thread which got closed without me having any answer to it. I've found it particularly painful to work with influxdb, and I am now on the brink of switching definitively to another time series db, although I would have like to give influxdb a fair chance.

The problem is that after many inserts trials, influxdb is just getting blocked with no possible access:
1) I cannot connect to the database via python:
  File "/home/ezcocos/dev/python/pyenv2_7/local/lib/python2.7/site-packages/requests/adapters.py", line 407, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', error(111, 'Connection refused'))

2) I cannot access either the admin tools via localhost:8083 -> page not available

My only solution so far was to reinstall influxdb from scratch, what I will not be able to do in production. My question is, is there a way at least to repair influxdb in order to be able to access it again without completely reinstalling it loosing all the data?

I run influxdb on ubuntu 14.10.

Many thanks in advance for your help.
ezcocos

**Comment by ybizeul** [comment]
Do anyone has a workaround ? I can't think this is happening for everyone, there has to be something that triggers it that is specific to our installations

**Comment by toddboom** [comment]
@ybizeul It sounds like maybe the daemon isn't starting up at all. Are you able to successfully write data or execute queries when it's in this state?

**Comment by m1keil** [comment]
We are hitting same issue with older InfluxDB version (0.8.0) and it looks like it is the same in 0.8.8.

If I delete `data/raft` folder and restart InfluxDB daemon it starts up with 8083 & 8086 ports open.
It looks like raft isn't tolerating the changes of hostnames (which happens when you kill/start new containers). I noticed it in the logs and you can see it in @kureikain logs as well:

```
[2014/12/02 19:32:49 UTC] [INFO] (github.com/influxdb/influxdb/server.(*Server).ListenAndServe:112) Sending change connection string command (d98cd1d00f41:8099,f4fe021fe272:8099) (http://d98cd1d00f41:8090,http://f4fe021fe272:8090)
```

`d98cd1d00f41` is the current container, while `d98cd1d00f41` is the old one (which is no longer exists). Any chance this is due to Raft's election process and the fact that raft cannot work with 2 servers (it needs 3 or more)?

**Comment by DragosCoraid** [comment]
@m1keil that's an interesting theory. It seems to be random though, sometimes stopping a container and restarting it works, other times it does not. Once it stops working, it's done, the only way I can recover from it is to delete the data.

Our container uses supervisord to manage the influxdb process + other processes. supervisord is PID 1 and we stop the container by using "docker stop" which sends a SIGTERM to supervisord which in turn sends a SIGTERM to the processes it manages. Docker will wait by default for 10 sec. before sending a SIGKILL if the container didn't stop as a result of SIGTERM. We tried increasing the time between SIGTERM and SIGKILL thinking that maybe influx didn't get enough time to shut down gracefully, but that still does not help.

**Comment by m1keil** [comment]
There's definitely something "random" going on. It's very easy to reproduce this with the following Dockerfile:

```
FROM phusion/baseimage:0.9.15

RUN curl -O http://s3.amazonaws.com/influxdb/influxdb_0.8.8_amd64.deb && \
    dpkg -i influxdb_0.8.8_amd64.deb && \
    rm -rf influxdb_0.8.8_amd64.deb

RUN mkdir -p /etc/service/influxdb && \
    echo '#!/bin/sh' > /etc/service/influxdb/run && \
    echo 'exec /usr/bin/influxdb -config=/opt/influxdb/shared/config.toml' >> /etc/service/influxdb/run && \
    chmod +x /etc/service/influxdb/run
``` 
This will build small container image with InfluxDB and runit as the init system. Runit will monitor Influx's state and auto start it as soon as it is down. 
To build it: `docker build -t influxdb <path to Dockerfile>`

## Case 1
### Run Docker container without static hostname
`docker run -t -d --name="influx" -v <local dir>:/opt/influxdb/shared/data influxdb /sbin/my_init`
This should start container and populate all Influx data in `<local dir>`.

Now kill the container and start another one again:
```
$ docker rm -f influx
$ docker run -t -d --name="influx" -v /home/vagrant/influxdata:/opt/influxdb/shared/data influxdb /sbin/my_init
```

Execute bash process inside the container and inspect its state:
```
$ docker exec -it influx bash
root@901f13fc06b9:/# ps -ef | grep influx
root       106   102  0 17:38 ?        00:00:00 runsv influxdb
root       107   106  0 17:38 ?        00:00:00 /usr/bin/influxdb -config=/opt/influxdb/shared/config.toml
root       136   120  0 17:39 ?        00:00:00 grep --color=auto influx
root@901f13fc06b9:/#  ss -tln
State       Recv-Q Send-Q                                                          Local Address:Port                                                            Peer Address:Port
LISTEN      0      128                                                                         *:22                                                                         *:*
LISTEN      0      128                                                                        :::8099                                                                      :::*
LISTEN      0      128                                                                        :::8083                                                                      :::*
LISTEN      0      128                                                                        :::8086                                                                      :::*
LISTEN      0      128                                                                        :::22                                                                        :::*
LISTEN      0      128
```

Now send SIGTERM to influxdb process:
```
root@901f13fc06b9:/# killall -15 influxdb
.... wait 30 seconds ....
root@901f13fc06b9:/# ss -tln
State       Recv-Q Send-Q                                                          Local Address:Port                                                            Peer Address:Port
LISTEN      0      128                                                                         *:22                                                                         *:*
LISTEN      0      128                                                                        :::22                                                                        :::*
LISTEN      0      128                                                                        :::8090                                                                      :::*
```
Sometimes it will recover after first SIGTERM, but additional SIGTERM will do the job. That's the random part. It seems like the only way to recover is to delete raft's folder:
```
root@901f13fc06b9:/# rm -rf /opt/influxdb/shared/data/raft
root@901f13fc06b9:/# killall -15 influxdb
... wait few seconds ..
root@901f13fc06b9:/# ss -tln
State       Recv-Q Send-Q                                                          Local Address:Port                                                            Peer Address:Port
LISTEN      0      128                                                                         *:22                                                                         *:*
LISTEN      0      128                                                                        :::8099                                                                      :::*
LISTEN      0      128                                                                        :::8083                                                                      :::*
LISTEN      0      128                                                                        :::8086                                                                      :::*
LISTEN      0      128                                                                        :::22                                                                        :::*
LISTEN      0      128                                                                        :::8090                                                                      :::*
````
## Case 2 
### Run Docker container with static hostname
Remove previous running container and delete Influx's data files from `<local dir>`.

Now repeat the same steps as in Case 1, only now launch container instances with static hostname:
`$ docker run -t -d --name="influx" --hostname="influxdb" -v /home/vagrant/influxdata:/opt/influxdb/shared/data influxdb /sbin/my_init`

You'll see that no matter how much SIGTERM you send in its way, it will successfully recover after few seconds.

**Comment by entone** [comment]
using fig and/or docker-compose I was running into this same issue, I thought it was a data volume issue initially, but setting the hostname as mentioned fixed the problem.

**Comment by pauldix** [comment]
This will be fixed in 0.9.0. We won't be making any new releases in the 0.8 line. In the meantime, set the hostname in your config file and it should work across restarts.

**Comment by bercab** [comment]
If helps someone running docker, we have been able to recover influxdb from this issue by running docker with same original hostname. 

In fact, this has been the only way to recover maintaining the old data. 

You can extract the original docker hostname (container id) from *raft/log* binary log on your data dir: 

Binary editing the file, you can get something like:
```
...
{"name":"a4ba0aaf1b324944","connectionString":"http://03fba15f761d:8090","protobufConnectionString":"03fba15f761d:8099"}
...
```

So the id in *connectionString" is your old docker id.

Recover with: 
```
docker run --hostname=03fba15f761d  ...
```

**Comment by beckettsean** [comment]
@gunnaraasen to note re: Docker

**Comment by aboltart** [comment]
@bercab Thanks, your comment helped

**Comment by AurelienLourot** [comment]
This issue was closed because it was planned to be fixed in 0.9.0 according to @pauldix, but the issue is still there in 0.9.4.2 and I can't find anything related in the [changelog](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md). Will this issue be reopened?

**Comment by beckettsean** [comment]
@AurelienLourot InfluxDB 0.9 is no longer receiving code updates. There will be no fixes to the 0.9.4.2 code base. I would recommend upgrading to InfluxDB 0.9.6 to see if that helps.

Also, please note that all other reports in this issue are for InfluxDB 0.8.x, so it is likely that while your symptoms appear similar it's not actually the same underlying cause. I encourage you to email the mailing list at influxdb@googlegroups.com for assistance.
